driver-examples/tycho/com.datastax.driver.examples.rcp.mailbox.tests/src/main/java/com/datastax/driver/examples/rcp/mailbox/tests/CassandraRule.java

```diff
@@ -0,0 +1,65 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.examples.rcp.mailbox.tests;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import org.junit.rules.ExternalResource;
+
+import com.datastax.driver.core.CCMBridge;
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Session;
+
+public class CassandraRule extends ExternalResource {
```

Comment:
This JUnit rule can then be used either as a `@ClassRule` or a `@Rule`  to initialize a cluster and a session on a per-class or per-test basis respectively. See also below my comments for `CassandraServer`.